account_invoice_merge/models/account_move.py

```diff
@@ -64,10 +66,39 @@ def _get_first_invoice_fields(self, invoice):
             "partner_bank_id": invoice.partner_bank_id.id,
         }
 
+    @api.model
+    def _get_sum_fields(self):
+        return ["quantity"]
+
+    @api.model
+    def _get_invoice_line_vals(self, line):
+        field_names = self._get_invoice_line_key_cols() + self._get_sum_fields()
+        vals = {}
+        origin_vals = line._convert_to_write(line._cache)
+        for field_name, val in origin_vals.items():
+            if field_name in field_names:
+                vals[field_name] = val
+        return vals
+
     def _get_draft_invoices(self):
         """Overridable function to return draft invoices to merge"""
         return self.filtered(lambda x: x.state == "draft")
 
+    def make_key(self, br, fields):
```

Comment:
this method should be tested.
For a given record:
`["m2o_field", "m2m_field", "o2m_field", "char_field", ...]`
Should return something like that
```python
{
"m2o_field": ID,
"m2m_field": [fields.Commands.set(IDS)],
"o2m_field": [fields.Commands.set(IDS)],
"char_field": VALUE,
…
}
```

Reply:
No: it does not care the type of fields but the type of values.
Yes: i will add the test for 3 types: Dict, Model, Other

Reply:
I had to check… Looks like odoo became very permissive about how to write on relational fields!
```python
>>> so
sale.order(1,)
>>> so2
sale.order(2,)
>>> so.order_line
sale.order.line(1, 2)
>>> lines = so2.order_line
>>> lines
sale.order.line(3, 4, 5)
>>> so.write({"order_line": [fields.Command.set(lines.ids)]})
True
>>> so.write({"order_line": tuple(lines.ids)})
True
>>> so.write({"partner_id": so2.partner_id.id})
True
>>> so.write({"partner_id": tuple(so2.partner_id.ids)})
True
```
My bad, I wasn't aware about this… :pray:

Reply:
And I was also confused about the usage of this make key method. Sorry about that